### PR TITLE
Make nodes clickable

### PR DESCRIPTION
**Pull request by JaspervanTilburg**

### CODE REVIEW

File: src/main/java/graph/SequenceGraph.java [hunk 1]
```diff
@@ -27,6 +27,10 @@ public SequenceGraph() {
         this.edges = new ArrayList<Edge>();
     }
 
+    public int size() {
+        return nodes.size();
+    }
```
Comment (Jipje) [inline src/main/java/graph/SequenceGraph.java]: Did you test this method?

File: src/main/java/gui/GraphDrawer.java [hunk 2]
```diff
@@ -82,7 +90,7 @@ void moveShapes(final double xDifference) {
         gc.clearRect(0, 0, gc.getCanvas().getWidth(), gc.getCanvas().getHeight());
         gc.setFill(Color.BLUE);
 
-        for (int j = 0; j < columns.size(); j++) {
+        /*for (int j = 0; j < columns.size(); j++) {
             ArrayList<SequenceNode> column = columns.get(j);
             for (int i = 0; i < column.size(); i++) {
                 if (column.get(i).isDummy()) {
```
Comment (Jipje) [inline src/main/java/gui/GraphDrawer.java]: Maybe remove this commented code.

File: src/main/java/gui/GraphDrawer.java [hunk 3]
```diff
@@ -119,11 +148,23 @@ private void drawEdges(double xDifference) {
         }
     }
 
+    public SequenceNode clickNode(double xEvent, double yEvent) {
+        SequenceNode click = null;
+        for (int i = 0; i < canvasNodes.size(); i++) {
+            DrawableNode node = canvasNodes.get(i);
+            node.lowlight();
+            if (node.checkClick(xEvent, yEvent)) {
```
Comment (Jipje) [inline src/main/java/gui/GraphDrawer.java]: Looks good

File: src/main/java/gui/MenuController.java [hunk 4]
```diff
@@ -145,6 +146,11 @@ public void zoomOutClicked() throws IOException {
     @FXML
     public void clickMouse(MouseEvent mouseEvent) {
         pressedX = mouseEvent.getX();
+        double y = mouseEvent.getY();
+        SequenceNode clicked = drawer.clickNode(pressedX, y);
+        if (clicked != null) {
```
Comment (Jipje) [inline src/main/java/gui/MenuController.java]: Awesome.

File: src/main/java/gui/DrawableNode.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
+package gui;
+
+import com.sun.corba.se.impl.orbutil.graph.Graph;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.paint.Color;
+
+/**
+ * Created by TUDelft SID on 17-5-2017.
+ */
+public class DrawableNode {
```
Comment (Jipje) [inline src/main/java/gui/DrawableNode.java]: I feel like we should test this class.

Comment (JaspervanTilburg) [inline src/main/java/gui/DrawableNode.java]: Tested this class partly, found some trouble testing with the GraphicContext class.

File: src/main/java/graph/SequenceGraph.java (hunk 1 above)
Comment (JaspervanTilburg) [inline src/main/java/graph/SequenceGraph.java]: Method tested :)

File: src/main/java/gui/GraphDrawer.java (hunk 2 above)
Comment (JaspervanTilburg) [inline src/main/java/gui/GraphDrawer.java]: I removed it, is indeed not needed anymore

**Comment by Jipje** [comment]
Awesome, if Travis builds, go your gang and merge.

**Pull request merged**